packages/opentelemetry-tracing/src/BasicTracerRegistry.ts

```diff
@@ -0,0 +1,60 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ConsoleLogger } from '@opentelemetry/core';
+import * as types from '@opentelemetry/types';
+import { Logger } from '@opentelemetry/types';
+import { SpanProcessor, Tracer } from '.';
+import { DEFAULT_CONFIG } from './config';
+import { MultiSpanProcessor } from './MultiSpanProcessor';
+import { NoopSpanProcessor } from './NoopSpanProcessor';
+import { TracerConfig } from './types';
+
+export class BasicTracerRegistry implements types.TracerRegistry {
+  private readonly _registeredSpanProcessors: SpanProcessor[] = [];
+  private readonly _tracers: Map<string, Tracer> = new Map();
+
+  activeSpanProcessor = new NoopSpanProcessor();
+  readonly logger: Logger;
+
+  constructor(private _config: TracerConfig = DEFAULT_CONFIG) {
+    this.logger = _config.logger || new ConsoleLogger(_config.logLevel);
+  }
+
+  getTracer(name = '', version = '*', config?: TracerConfig): Tracer {
```

Comment:
How do you stop particular tracer? For example: if the `mysql` plugin suddenly started to misbehave
 by creating either bogus or a large number of telemetry data.

Reply:
There is currently no mechanism to stop a given tracer. that is apparently coming later according to spec